docs/DOCKER.md

```diff
@@ -0,0 +1,83 @@
+
+This allows to set up a development environment for CyclOSM easily. Specifically, this environment consists of a
+PostgreSQL database to store the OpenStreetMap data and [Kosmtik](https://github.com/kosmtik/kosmtik) for previewing the style.
+
+## Prerequisites
+
+Docker is available for Linux, macOS and Windows. [Install](https://www.docker.com/get-docker) the software packaged for your host system in order
+to be able to run Docker containers. You also need Docker Compose, which should be available once you installed
+Docker itself. Otherwise you need to [install Docker Compose manually](https://docs.docker.com/compose/install/).
+
+You need sufficient disk space of _several Gigabytes_. Docker creates a disk image for its virtual machine that holds the virtualised operating system and the containers. The format (Docker.raw, Docker.qcow2, \*.vhdx, etc.) depends on the host system. It can be a sparse file allocating large amounts of disk space, but still the physical size starts with 2-3 GB for the virtual OS and grows to 6-7 GB when filled with the containers needed for the database, Kosmtik, and a small OSM region. Further 1-2 GB are needed for shape files in the cyclosm/data repository.
+
+## Quick start
+
+If you are eager to get started here is an overview over the necessary steps.
+Read on below to get the details.
+
+* `git clone https://github.com/cyclosm/cyclosm-cartocss-style.git` to clone CyclOSM repository into a directory on your host system
+* download OpenStreetMap data in osm.pbf format to a file `data.osm.pbf` and place it within the CyclOSM directory (for example some small area from [Geofabrik](https://download.geofabrik.de/))
+* If necessary, `sudo service postgresql stop` to make sure you don't have currently running a native PostgreSQL server which would conflict with Docker's PostgreSQL server.
+* `docker-compose up import` to import the data (only necessary the first time or when you change the data file)
+* `docker-compose up kosmtik` to run the style preview application
+* browse to [http://localhost:6789](http://localhost:6789) to view the output of Kosmtik
+* Ctrl+C to stop the style preview application
+* `docker-compose stop db` to stop the database container
+
+## Repositories
+
+Instructions above will clone main CyclOSM repository. To test your own changes you should [fork](https://help.github.com/articles/fork-a-repo/) cyclosm/cyclosm-cartocss-style repository and [clone your fork](https://help.github.com/articles/cloning-a-repository/).
+
+This CyclOSM repository needs to be a directory that is shared between your host system and the Docker virtual machine. Home directories are shared by default; if your repository is in another place you need to add this to the Docker sharing list (e.g. macOS: Docker Preferences > File Sharing; Windows: Docker Settings > Shared Drives).
+
+## Importing data
```

Comment:
This might need some extra info on downloading shapefiles & elevation, and generating contours and hillshading.